server/encoder/video_encoder_vulkan.cpp

```diff
@@ -667,15 +675,40 @@ std::pair<bool, vk::Semaphore> wivrn::video_encoder_vulkan::present_image(vk::Im
 	{
 		slot_item.idr = false;
 	}
+
+	// avoid self-reference when the DPB ring wraps (cannot reference the slot we're writing).
+	bool use_ref = (ref_slot && ref_slot->info.slotIndex != static_cast<int32_t>(slot_index));
+
```

Comment:
It looks like we could remove the bool, use `ref_slot`  (implicitly `!= nullptr`) instead by moving the `slotIndex` check in the `if (i.frame_index == last_ack and i.info.slotIndex != -1)` condition.
The condition becomes: reference slot is valid, not selected for overwrite and acknowledged by client.